pallets/collator-selection/src/lib.rs

```diff
@@ -499,29 +512,86 @@ pub mod pallet {
 			collators.extend(candidates.into_iter().collect::<Vec<_>>());
 			collators
 		}
-		/// Kicks out and candidates that did not produce a block in the kick threshold.
+
+		/// Removes collators with unsatisfactory performance
+		/// Returns the removed AccountIds
 		pub fn kick_stale_candidates(
 			candidates: Vec<CandidateInfo<T::AccountId, BalanceOf<T>>>,
 		) -> Vec<T::AccountId> {
-			let now = frame_system::Pallet::<T>::block_number();
-			let kick_threshold = T::KickThreshold::get();
-			candidates
+			// 0. Sanity checks, no performance recorded or candidates, no kicking
+			let mut collator_perf_this_session =
+				<BlocksPerCollatorThisSession<T>>::iter().collect::<Vec<_>>();
+			if collator_perf_this_session.is_empty() {
+				return Vec::new();
+			}
+			// 1. Sort collator performance list
+			collator_perf_this_session.sort_unstable_by_key(|k| k.1); // XXX: don't like the tuple accessor, could this be a struct?
+														  // collator_perf_this_session.reverse();
+			let no_of_candidates = collator_perf_this_session.len();
+
+			// 2. get percentile by _exclusive_ nearest rank method https://en.wikipedia.org/wiki/Percentile#The_nearest-rank_method (rust percentile API is feature gated)
+			let ordinal_rank = (num_traits::float::FloatCore::ceil(
```

Comment:
There is actually one "problem" with this algorithm in that for validator counts below 5 it uses the highest performing validator for the percentile. That's not necessarily a bad thing, but something to keep in mind, as for low validator counts this approach is equivalent to "kick everyone 10% worse than the best collator" ( unless invunerable )

I think we already run 5+ nodes on calamari, so it shouldn't be an issue

![image](https://user-images.githubusercontent.com/4237655/152894100-43e9014d-40df-4c9b-819f-5722b4754a7f.png)